### PR TITLE
SQLDB notebook merge into sqlvmdeployments

### PR DESCRIPTION
**Pull request by smartguest**
This PR adds the current version of SQL DB notebook into @aasimkhan30 's sqlvmdeployments2 branch.

(NOTE: The files for deployAzureSQLVMWizard are the most recent ones from the aasim/sqlvmdeployments2 branch. For some reason, the pull request shows that they're overwriting older versions of the files, but they are unchanged from the main branch.)

### CODE REVIEW

File: extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/deployAzureSQLDBWizard.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		const pages: WizardPageBase<DeployAzureSQLDBWizard>[] = [];
+		pages.push(new AzureSettingsPage(this));
+		pages.push(new DatabaseSettingsPage(this));
+		pages.push(new AzureSQLDBSummaryPage(this));
+		return pages;
+	}
+
+	private async scriptToNotebook(): Promise<void> {
+		const variableValueStatements = this.model.getCodeCellContentForNotebook();
+		const insertionPosition = 2; // Cell number 5 is the position where the python variable setting statements need to be inserted in this.wizardInfo.notebook.
```
Comment (aasimkhan30) [inline extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/deployAzureSQLDBWizard.ts:61]: Correct the comment

File: extensions/resource-deployment/src/ui/deployAzureSQLVMWizard/constants.ts [hunk 2]
```diff
@@ -45,4 +45,10 @@ export const NetworkSettingsNewPublicIp = localize('deployAzureSQLVM.NetworkSett
 export const RDPAllowCheckboxLabel = localize('deployAzureSQLVM.VmRDPAllowCheckboxLabel', "Enable RDP(3389) inbound Port");
 
 // SQL Server settings page constants
-export const SqlServerSettingsPageTitle = localize('deployAzureSQLVM', "SQL Servers settings");
+export const SqlServerSettingsPageTitle = localize('deployAzureSQLVM.SqlServerSettingsPageTitle', "SQL Servers settings");
```
Comment (aasimkhan30) [inline extensions/resource-deployment/src/ui/deployAzureSQLVMWizard/constants.ts:48]: Can you discard changes in SQL VM wizard

File: extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/pages/databaseSettingsPage.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+			this.liveFormValidation();
+		});
+
+		this._startIpAddressTextRow = this.wizard.createFormRowComponent(view, constants.StartIpAddressLabel, '', this._startIpAddressTextbox, true);
+
+		//End IP Address Section:
+
+		this._endIpAddressTextbox = view.modelBuilder.inputBox().withProperties(<azdata.InputBoxProperties>{
+			inputType: 'text',
+			value: '0.0.0.0'
```
[the remark this answers is not in the document]
Comment (smartguest) [reply extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/pages/databaseSettingsPage.ts:108]: Same default values as on the create-sqldb notebook. Portal does not give fields to edit a firewall rule. Default is to allow all azure internal IP addresses access.

File: extensions/resource-deployment/src/services/resourceTypeService.ts [hunk 4]
```diff
@@ -279,6 +284,9 @@ export class ResourceTypeService implements IResourceTypeService {
 		} else if (instanceOfAzureSQLVMDeploymentProvider(provider)) {
 			const wizard = new DeployAzureSQLVMWizard(provider.azureSQLVMWizard, this.notebookService, this.toolsService);
 			wizard.open();
+		} else if (instanceOfAzureSQLDBDeploymentProvider(provider)) {
+			const wizard = new DeployAzureSQLDBWizard(provider.azureSQLDBWizard, this.notebookService, this.toolsService);
+			wizard.open();
```
Comment (aaomidi) [inline extensions/resource-deployment/src/services/resourceTypeService.ts:289]: This is a promise, looks like theres also some problems in the base code too.

File: extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/deployAzureSQLDBWizard.ts [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+	public get notebookService(): INotebookService {
+		return this._notebookService;
+	}
+
+	public get toolService(): IToolsService {
+		return this._toolsService;
+	}
+
+	constructor(private wizardInfo: AzureSQLDBWizardInfo, private _notebookService: INotebookService, private _toolsService: IToolsService) {
```
Comment (aaomidi) [inline extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/deployAzureSQLDBWizard.ts:30]: nit: Put the constructor as the first thing in the class before the getters

File: extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/deployAzureSQLDBWizardModel.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+		statements.push(`azure_sqldb_nb_var_resource_group_name = '${this.azureResouceGroup}'`);
+		statements.push(`azure_sqldb_server_name = '${this.azureServerName}'`);
+		statements.push(`azure_sqldb_database_name = '${this.databaseName}'`);
+		statements.push(`azure_sqldb_location = '${this.azureRegion}'`);
+		statements.push(`azure_sqldb_ip_start = '${this.startIpAddress}'`);
+		statements.push(`azure_sqldb_ip_end = '${this.endIpAddress}'`);
+		statements.push(`azure_sqldb_firewall_name = '${this.firewallRuleName}'`);
+		//statements.push(`azure_sqldb_new_server = '${this.newServer}'`); //@todo alma1 9/8/2020 used for upcoming server creation feature.
+
+		return statements.map(line => line + EOL);
```
Comment (aaomidi) [inline extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/deployAzureSQLDBWizardModel.ts:43]: Don't use + for strings, can have unexpected behavior

File: extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/deployAzureSQLDBWizard.ts (hunk 5 above)
Comment (smartguest) [reply extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/deployAzureSQLDBWizard.ts:30]: This is following @aasimkhan30 's code to be exact, along with the other lines, but I'll change it once I notify Aasim about it.

**Pull request closed without merging**